### ISSUE TITLE
Show and Tell: A Neural Image Caption Generator

### ISSUE THREAD

**Issue by wrryu09** [GDSC-Ewha-4th/Study-paperReview#56]
# Show and Tell: A Neural Image Caption Generator

#
### Abstract

**NIC, end-to-end neural network system that can automatically view an image and generate a reasonable description in plain English !!!**

present a generative model based on a deep recurrent architecture that combines recent advances in computer vision and machine translation and that can be used to **generate natural sentences describing an image**.

이미지 내용 설명 → 컴퓨터 비전과 자연어처리 연결 인공지능

이미지를 넣었을 때 이를 설명하는 문장의 유사성을 최대화하도록 트레이닝됨
#

### 1. Introduction

description must capture not only the objects contained in an image, but it also must express how these objects related to each other as well as their attributes and the activities they are involved in.

visual understanding + 자연어 처리를 위한 language model도 필요

기존 시도들 : 이미지에서 그에 따른 설명을 만들어내기 위해 존재하던 sub-problem의 솔루션을 합침

이미지 I를 인풋으로 하고 그 이미지를 적절하게 설명하는 단어 S*t*가 나오는 target sequence of words S={S1, S2, …}를 제공하는 likelihood p(S|I)를 최대화하도록 트레이닝된 single joint model을 만들어보고자 함

**RNN** Recurrent Neural Networks

“encoder” RNN이 source sentence를 읽고 rich fixed-length vector representation으로 변형하면, 이가 타깃 문장을 생성하는”decoder” 의 초기 hidden state로 사용됨.

encoder RNN을 deep convolution neural network(CNN)로 바꿔보자

CNN을 미리 이미지 분류 task로 트레이닝을 함으로써 encoder로 사용하고, 이의 last hidden layer를 문장을 생성하는 RNN decoder의 인풋으로 사용  → Neural Image Captrion, NIC

1. end-to-end system
    
    trainable using stochastic gradient descent
    
2. combine state-of-art sub-networks for vision and language models
    
    larger corpora 에서 pre-trained 될 수 있고 추가적인 데이터의 장점을 가져갈 수 있음
    
3. 다른 접근들에 비해 좋은 performance ~

#

### 2. Related Work

들과 다른 점

we use a more powerful RNN model, and provide the visual input to the RNN model directly, which makes it possible for the RNN to keep track of the objects that have been explained by the text
#

### 3. Model

~we propose a neural and probabilistic framework to generate descriptions from images

강력한 시퀀스 모델 사용하면 end-to-end형식 가진 인풋 문장에서 올바른 translation의 가능성을 즉각적으로 최대화 가능

이 모델 encoder RNN( variable length input → fixed demensional vector로 인코딩) 만들고 decode 할 때도 사용

To make RNN more concrete

- what is the exact form of f
- how are the images and words fed as inputs x*t*

### 3.1. LSTM-based Sentence Generator
![image](https://user-images.githubusercontent.com/98469609/202906993-bc6c5f0d-b5ff-4b8a-94cf-e7ed38bd1f2f.png)

**L**ong **S**hort **T**erm **M**emory

vanishing, exploding gradients를 다루기 위해 LSTM 도입, translation, sequence generation 잘 했다

memory cell **c** : core of LSTM model

encoding knowledge at every time step of what inputs have been observed up to this step

인풋 읽고 새로운 셀 값 아웃 할지말지 결정
#

### 4.1. Evaluation Metrics

생성된 description이 적절한지 평가하기 위한 평가 metric 만듦

### 4.2. Datasets

Pascal VOC 2008, Flickr8k, Flickr30k, MSCOCO, SBU 등을 train, valid, test 셋으로 나누어 사용

각기 다른 데이터셋을 사용하기 위한 방법들 설명

### 4.3. Results

how dataset size affects generalization, what kinds of transfer learning it would be able to achieve, 알고싶었다

### 4.3.1 Training Details

overfitting이 문제였고 assigning description task가 힘들었음

overfitting 해결 방안: CNN component weight initializing to a pretrained model

### 4.3.2 Generation Results

BLEU 점수 등

### 4.3.3 Transfer Learning, Data Size and Label Quality

training data 추가할수록 결과 좋아짐(Flickr30, Flickr8k간 데이터 사이즈 비교)

MSCOCO는 훨씬 더 큰 데이터셋이지만 collection process가 다르게 되어 있어서 BLEU 점수 떨어짐

PASCAL, SBU 등 결과

### 4.3.4 Generation Diversity Discussion

모델이 novel caption 생성하는가? caption이 다양하고 품질이 좋은가?

생성된 캡션들 중 best candidate를 뽑으면 80%정도는 training set에 있는 거여서 대신 top 15 generated sentence 분석하면 반 정도는 괜찮은 novel description을 볼 수 있었음

### 4.3.6 Human Evaluation

BLEU가 NIC와 인간이 쓴 설명 차이를 잘 잡아내는 것을 보니 완벽한 메트릭이 아니다

![image](https://user-images.githubusercontent.com/98469609/202907034-33c0c4cb-ae35-4ea4-b1b6-468a780fd40d.png)
#

### 4.3.7 Analysis of Embeddings

이전 단어 S*t-1*을 S*t*를 만들어내는 decoding LSTM에 넣기 위해서 word embedding vectors를 썼었다

independent of the size of the dictionary 의 장점을 가짐

나머지 모델들과도 공통적으로 트레이닝 가능

어떻게 언어의 통계에서 learned representations가 의미를 잡아냈을까? → the nearest other words found in the learned embedding space

horse pony donkey 같은 서로 비슷한 단어들은 CNN이 ‘말같이 생긴’ 동물들과 연관이 있는 요소들을 내보내도록 함

유니콘같은 극단적인 케이스는 기존 단어들에서 온 접근을 잃어버리게 하기 위해서 추가적인 정보가 많이 필요할 듯
#
### 5. Conclusion

**NIC, end-to-end neural network system that can automatically view an image and generate a reasonable description in plain English**

를 만들었다!

NIC는 이미지를 compact representation으로 인코딩하는 CNN, 상응하는 문장을 생성하는 RNN에 기반

모델은 주어진 이미지에 대한 문장의 likelihood를 최대화하도록 트레이닝됨

NIC를 평가하기 위해 몇가지 데이터셋으로 실험해봤는데 생성된 문장의 질을 평가하기 위해 BLEU를 사용했다

datasets size가 증가할수록 NIC performance도 증가할 것이다

**Comment by yj-song131** [comment]
자연어 처리와 컴퓨터 비전을 연결한다니 흥미롭네요! 잘 읽었습니다 ㅎㅎ

**Comment by minha62** [comment]
image captioning 궁금했었는데 자세하게 정리되어 있어서 보기 좋았습니다! 좋은 내용 감사합니다:)
